### PR TITLE
feat(widget-resize): add size select field

### PR DESCRIPTION
**Pull request by narsaynorath**
Adds a feature-flagged, hard-coded select field for later manipulation of widget sizes.

Screen Shot 2021-11-16 at 12 10 14 PM

Note: Not sure yet if I should represent the values as enums or if it's okay with strings. I was thinking of hashing that out in the work where I actually begin using this select field.

### CODE REVIEW

**Review by shruthilayaj — COMMENTED** [review]
Is it possible to reuse the existing `SelectControl` (?) component and align it with the other icons in that line? Like use something like this so it doesn't seem out of place?
Screen Shot 2021-11-16 at 12 28 48 PM

**Comment by narsaynorath** [comment]
@shruthilayaj I'd love to use that dropdown component. I took a quick look around the code and I don't think it's implemented as a reusable component (please correct me if I'm wrong because I agree with you). I did look at the storybook and I found some Select Field stuff that looks better. There seems to be a z-index issue though

Screen Shot 2021-11-16 at 1 20 07 PM

If we expect to replace this with a draggable corner kind of UI in EA, would it be fine to keep this super basic UI for now?

File: static/app/views/dashboardsV2/widgetCard.tsx [hunk 1]
```diff
@@ -81,6 +83,11 @@ class WidgetCard extends React.Component<Props> {
 
     return (
       <ToolbarPanel>
+        <Feature features={['organizations:dashboard-widget-resizing']}>
+          <SizeSelectContainer style={{visibility: hideToolbar ? 'hidden' : 'visible'}}>
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/widgetCard.tsx:87]: Noticed that my size selector wasn't being hidden on drag. I duplicated the visibility styling here to keep my changes low impact to the surrounding code, should I have made a container that surrounds both `Feature` and `IconContainer` instead for DRY-ness? 🤔

**Review by shruthilayaj — APPROVED** [review]
looks good, two minor comments- pull the selector component into the widgetCard file and remove the `organizations:` prefix when checking the feature flag.

File: static/app/views/dashboardsV2/widgetCard.tsx [hunk 2]
```diff
@@ -33,6 +35,27 @@ import {Widget} from './types';
 import WidgetCardChart from './widgetCardChart';
 import WidgetQueries from './widgetQueries';
 
+type SizeSelectorProps = {
+  size: string;
+  onSizeChange: (size: string) => void;
+};
+
+export const SizeSelector = ({size, onSizeChange}: SizeSelectorProps) => {
```
Comment (shruthilayaj) [inline static/app/views/dashboardsV2/widgetCard.tsx:43]: can remove this export if it's only being used in this file!

Comment (narsaynorath) [reply static/app/views/dashboardsV2/widgetCard.tsx]: Ah, thanks for catching that! Forgot to remove it when I was putting the component in the same file

**Pull request merged**